Develop

sylwiavv opened the pull request:
I'm having some issues with the handleEdit function. Since it works with asynchronous logic, the editTime value from Zustand is always one step behind and delayed compared to let value = e.target.value;.

Additionally, I need to implement increase and decrease management and fix search params when the timer is in edit mode.

sylwiavv committed:
WIP: preliminary styling edit timer component

sylwiavv committed:
WIP: move buttons from Timer, rewrite reset action

sylwiavv committed:
Add logic to set timer from URLSearchPArams

sylwiavv committed:
Add logic to handle errors

sylwiavv committed:
Some styles changes, add edit timer state

sylwiavv committed:
rm some comments

sylwiavv committed:
fix some build errors

jugeeer commented on src/hooks/useCountDown.tsx:
```diff
@@ -28,7 +26,6 @@ const useCountdown = ({
   const start = () => toggleRunning(true);
```
this  should update your suztand value timerState to Erunning,Running

jugeeer commented on src/hooks/useCountDown.tsx:
```diff
@@ -28,7 +26,6 @@ const useCountdown = ({
   const start = () => toggleRunning(true);
   const pause = () => toggleRunning(false);
```
this  should update your suztand value timerState to Erunning,Pause

jugeeer commented on src/hooks/useCountDown.tsx:
```diff
@@ -28,7 +26,6 @@ const useCountdown = ({
   const start = () => toggleRunning(true);
   const pause = () => toggleRunning(false);
   const restart = () => {
```
and this one should this  should update your suztand value timerState to Erunning,idle, and resetTimerValue to equel the value that in searchParams

jugeeer commented on src/hooks/useCountDown.tsx:
```diff
@@ -5,15 +5,13 @@ interface CountdownOptions {
-  reset,
 }: CountdownOptions) => {
   useEffect(() => {
     if (!running || seconds <= 0) return;
```
you don't need all props that u pass here as a prop to this hook at the top just get value from zustand timer and setTimer, and inside interval update timer value

sylwiavv committed:
Rm conole.log

jugeeer commented on src/app/page.tsx:
```diff
@@ -1,19 +1,48 @@
+      setRestartTime(secondsToSet);
+      setEditTime(() => ({ hours: 0, minutes: 0, seconds: 0 }));
+    }
+  }, []);
```
if no searchParams and if not value in storage so then set initial time to 6 hours in search and zustand

jugeeer commented on src/store/timerStore.ts:
```diff
@@ -1,49 +1,81 @@
-      hours: 0,
-      minutes: 0,
       seconds: 0,
+      restartTime: 0,
```
remove restartTime because restart it's a searchParams

jugeeer commented on src/store/timerStore.ts:
```diff
@@ -1,49 +1,81 @@
+  setRestartTime: (time: number) => void;
+  setEditTime: (
+    updateFn: (prevValues: TTimeInitialValues) => TTimeInitialValues
+  ) => void;
```
decide what to use seconds or editTime

jugeeer commented on src/components/templates/BigTimer.tsx:
```diff
@@ -0,0 +1,180 @@
+            Restart
+          </Button>
+        )}
+      </ButtonsLayout>
```
move Start Reset and Puse to timer and create another Start

jugeeer commented on src/components/templates/BigTimer.tsx:
```diff
@@ -0,0 +1,180 @@
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [isEditingMode]);
```
move it inside EditTimer ant it's only should setEditingMode to false

jugeeer commented on src/components/templates/BigTimer.tsx:
```diff
@@ -0,0 +1,180 @@
+    } else {
+      params.set('seconds', hours.toString());
+      window.history.pushState(null, '', `?${params.toString()}`);
+    }
```
create another hook like useTimerSearchParams alos move this function to EditTimer

sylwiavv replied:
I already have setTimerSearchParams, can it be leave like that? (big-time/src/utils/setTimerSearchParams.tsx)

jugeeer commented on src/components/templates/BigTimer.tsx:
```diff
@@ -0,0 +1,180 @@
+    setSeconds(restartTime);
+    restart();
+    toggleRunning(ERunning.IDLE);
+  };
```
don't need this callback just use restart from useCountDown

jugeeer commented on src/components/templates/BigTimer.tsx:
```diff
@@ -0,0 +1,180 @@
+    start();
+  };
+
+  const hadnleEditMode = () => {
```
const hadnleEditMode = (value: boolen) => {

}

sylwiavv replied:
I think it can stay here because when timer is running after click on time the timer should stop, after second click edit timer component should open, what do you think?

jugeeer commented on src/components/templates/BigTimer.tsx:
```diff
@@ -0,0 +1,180 @@
+        editTimerRef.current &&
+        !editTimerRef.current.contains(event.target as Node)
+      ) {
+        setIsEditingMode(false);
```
hadnleEditMode(false)

jugeeer commented on src/components/templates/BigTimer.tsx:
```diff
@@ -0,0 +1,180 @@
+      <ButtonsLayout>
+        <CircleButtonWithIcon
+          icon={<Plus />}
+          onClick={() => {
```
just an single callback for both of them to update 
searchParams and timer form zustand
argiment like 'plus' | 'minus'

jugeeer commented on src/components/organisms/EditTimer/EditTimer.tsx:
```diff
@@ -1,62 +1,85 @@
+  // const { replace } = useRouter();
+
+  const [error, setError] = useState(false);
 
```
here u need a state your edit value

jugeeer commented on src/components/organisms/EditTimer/EditTimer.tsx:
```diff
@@ -1,62 +1,85 @@
+      setError(true);
+
+      setTimeout(() => {
+        setEditTime((prevValues) => ({
```
u don;t need to update zustadn u need to update the value from the state

sylwiavv committed:
fix paths

sylwiavv committed:
try to fix vercel err

sylwiavv committed:
Add some fixes from mr

sylwiavv committed:
Add refactoring

- rm unnecessary variables and fn from main store
- update search params after click "plus" and "minus"
- rewrite fn handleTimeChange for manage plus/minus clicking
- update useCountDown hook

sylwiavv committed:
Refactoring

- fix buttons logic
- fix logic after timer was edit
- update hook to set search params

sylwiavv committed:
Add all refactors from mr

sylwiavv committed:
fix paths

sylwiavv committed:
rm console.log

sylwiavv committed:
improve styles

jugeeer commented on src/app/constants/constants.tsx:
```diff
@@ -0,0 +1 @@
+export const sixHoursInSeconds = 21600;
```
unofficial rule to write constants in uppercase like SIX_HOURS_IN_SECONDS

jugeeer commented on src/app/page.tsx:
```diff
@@ -1,19 +1,46 @@
+      getSearchParamas({ searchParams, setSeconds });
+    }
+
+    let secondsFromLocalStorage;
```
don't use let at all, find a way to rewrite this logic without using let

jugeeer commented on src/store/timerStore.ts:
```diff
@@ -8,41 +9,43 @@ export enum ERunning {
       running: ERunning.IDLE,
       setTime: (value) =>
         set((state) => {
           let { seconds } = state;
```
don't use let

jugeeer commented on src/components/organisms/EditTimer/EditTimer.tsx:
```diff
@@ -1,61 +1,196 @@
+    });
+  };
+
+  const hadnleEditMode = (value: boolean) => {
```
no need to create such callback it's just update the type for `setIsEditingMode: (value: boolen) => void` and pss to this component just a callback rom useState

jugeeer commented on src/components/organisms/EditTimer/EditTimer.tsx:
```diff
@@ -1,61 +1,196 @@
+    e: React.ChangeEvent<HTMLInputElement>,
+    label: ETimerUnits
+  ) => {
+    let value = e.target.value;
```
don't use let

jugeeer commented on src/components/organisms/EditTimer/EditTimer.tsx:
```diff
@@ -1,61 +1,196 @@
+    if (error.error) return;
+
+    const hours =
+      editTime.find((item) => item.label === ETimerUnits.HOURS)?.value || 0;
```
you 3 the same find implement a callback that will accept 2 args item, unit

jugeeer commented on src/components/organisms/EditTimer/EditTimer.tsx:
```diff
@@ -1,61 +1,196 @@
+  };
+
+  if (!error.error) {
+  }
```
empty {}

sylwiavv committed:
WIP: add target as a search params

sylwiavv committed:
rm setTime

sylwiavv committed:
Add target to track timer, fit setSeconds from store

sylwiavv committed:
Add handleSetTarget into edit timer component

sylwiavv committed:
WIP: fix setting search params

sylwiavv committed:
Fix count down hook

sylwiavv committed:
Fixes

- add fn to stop coutnig down
- rm all lets

sylwiavv committed:
Add some refactors

split code into small parties

jugeeer commented on src/components/organisms/EditTimer/EditTimer.tsx:
```diff
@@ -1,61 +1,201 @@
+
+  updateFns.forEach((updateFn) => updateFn(params));
+  window.history.pushState(null, '', `?${params.toString()}`);
+};
```
you use this function at many places so that somethink common and shouldn't be exported from EditTImer move it to helper or utils

jugeeer commented on src/components/organisms/EditTimer/EditTimer.tsx:
```diff
@@ -1,61 +1,201 @@
+  label: ETimerUnits | null;
+};
+
+export const updateSearchParams = (
```
`const updateSearchParams = (hours, minutes, seconds, target) => {
    const params = new URLSearchParams(searchParams.toString())
    params.set(hours)
   params.set(minutes)
   params.set(seconds)
} `

jugeeer commented on src/utils/getSearchParams.tsx:
```diff
@@ -0,0 +1,30 @@
+  const minutesFromSearchParams = Number(params.get('minutes')) || 0;
+  const secondsFromSearchParams = Number(params.get('seconds')) || 0;
+
+  const miilli = convertToMilliseconds(
```
return convertToMilliseconds

jugeeer commented on src/hooks/useCountDown.tsx:
```diff
@@ -1,38 +1,61 @@
-import { useEffect } from 'react';
```
new hook 
useTimerSearchparams = () => {
   getValuesFromSearchParmas
   updateSearchParams
}

sylwiavv committed:
Refactoring

- create one hook to update and to get searchParams
- rm unnecceasy code and fn
- changed logic in page.tsx (handling local storage, search params)

sylwiavv committed:
Fix bug with URL updating

sylwiavv committed:
rm unnecessary file

sylwiavv committed:
Add some refactors

jugeeer commented on src/app/page.tsx:
```diff
@@ -1,19 +1,43 @@
+
+    if (searchParamsValues) {
+      const { hours, minutes, seconds, target } = searchParamsValues;
+      updateSearchParams({ hours, minutes, seconds, target });
```
fix here

jugeeer commented on src/app/page.tsx:
```diff
@@ -1,19 +1,43 @@
+    if (searchParamsValues) {
+      const { hours, minutes, seconds, target } = searchParamsValues;
+      updateSearchParams({ hours, minutes, seconds, target });
+    } else if (secondsFromLocalStorage === 0) {
```
if (secondsFromLocalStorage)

fix logic here

jugeeer commented on src/components/templates/BigTimer.tsx:
```diff
@@ -0,0 +1,55 @@
+    const target = event.target as HTMLElement;
+    const unit = target.dataset.unit;
+
+    if (!unit) return;
```
`if (!unit) {
  return
};`

fix logic here

jugeeer commented on src/components/organisms/EditTimer/EditTimer.tsx:
```diff
@@ -1,61 +1,171 @@
+  }, [isEditingMode]);
+
+  const handleSetStart = () => {
+    setIsEditingMode(false);
```
if (error,error) {
  return
}

jugeeer commented on src/components/organisms/Timer/Timer.tsx:
```diff
@@ -1,85 +1,55 @@
-          )}
-        </ButtonsLayout>
-      )}
+    if (remainingTime > 0) {
```
if (remainingTime)
fix logic

sylwiavv committed:
Add rest improvements from review

sylwiavv committed:
Revrite functions from useUpdateSearchParams into useCallback

sylwiavv committed:
Fix error with arr dependency
